### PR TITLE
Add Scala Native, update libraries, update scalafmt

### PR DESCRIPTION
**Pull request by lolgab**
- Update Scala 3 to `3.3.3`
- Update Scala 2.13 to `2.13.14`
- Update Scala 2.12 to `2.12.19`
- Update Scala.js to `1.16.0`
- Update Scala Native to `0.5.4`
- Update Scalafmt to `3.8.2`
- Update utest to `0.8.3`
- Update shapeless to `2.3.12`

### CODE REVIEW

File: boopickle/shared/src/test/scala/java/security/SecureRandom.scala [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package java.security
+
+// dummy unsecure SecureRandom implementation for tests
+// for real usage, use an implementation
+// like https://github.com/lolgab/scala-native-crypto
+class SecureRandom extends java.util.Random
```
Comment (cquiroz) [inline boopickle/shared/src/test/scala/java/security/SecureRandom.scala:6]: This is odd, do we need to include a class in the java namespace?
Is this only needed for native?

**Review by cquiroz — COMMENTED** [review]
Thanks for your PR.
I only checked the non formatting commit and in general it seems good
I had a question about the class in the java namespace

File: boopickle/shared/src/test/scala/java/security/SecureRandom.scala (hunk 1 above)
Comment (lolgab) [reply boopickle/shared/src/test/scala/java/security/SecureRandom.scala:6]: Yes, it's needed only on native, I need to move it to the right folder. I put it on `shared` by mistake.
I added it to not have to setup the openssl installation for tests.

Comment (lolgab) [reply boopickle/shared/src/test/scala/java/security/SecureRandom.scala]: Actually, it is needed also with Scala.js, so I added it to the `js-native` source directory.

**Comment by cquiroz** [comment]
Thanks a lot. LGTM

**Pull request merged**